apiserver/apiservermetrics.go

```diff
@@ -4,81 +4,88 @@
 package apiserver
 
 import (
-	"time"
-
+	"github.com/juju/juju/apiserver/observer/metricobserver"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 const (
-	apiserverMetricsNamespace = "juju_apiserver"
+	apiserverMetricsNamespace = "juju"
 )
 
-// ServerMetricsSource implementations provide apiserver metrics.
-type ServerMetricsSource interface {
-	TotalConnections() int64
-	ConnectionCount() int64
-	ConcurrentLoginAttempts() int64
-	ConnectionPauseTime() time.Duration
-}
+const (
+	SubsystemNamespace           = "apiserver"
+	DeprecatedSubsystemNamespace = "api"
+)
 
 // Collector is a prometheus.Collector that collects metrics based
 // on apiserver status.
 type Collector struct {
-	src ServerMetricsSource
-
-	connectionCounter        prometheus.Counter
-	connectionCountGauge     prometheus.Gauge
-	connectionPauseTimeGauge prometheus.Gauge
-	concurrentLoginsGauge    prometheus.Gauge
+	TotalConnections   prometheus.Counter
+	APIConnections     prometheus.Gauge
+	LogsinkConnections prometheus.Gauge
+	LoginAttempts      prometheus.Gauge
+	APIRequestsTotal   *prometheus.CounterVec
+	APIRequestDuration *prometheus.SummaryVec
 }
 
 // NewMetricsCollector returns a new Collector.
-func NewMetricsCollector(src ServerMetricsSource) *Collector {
+func NewMetricsCollector(subsystem string) *Collector {
```

Comment:
We don't want a subsystem here.  There are only two metrics that have the deprecated subsystem, and those are juju_api_request_duration_seconds and  juju_api_requests_total.

Actually looking at the data, the values for juju_api_requests_total are included in one of the sub values for juju_api_request_duration_seconds_count, so we should look to drop that one entirely in 2.6 and not create an apiserver subsystem for it.